### ISSUE TITLE
Load query results into Pydantic Object

### ISSUE THREAD

**Issue by derpy4me** [MarkyMan4/filequery#43]
## Description
I have been using Filequery for data files over the last couple of weeks, and one fallback that I've found is that it is not easy to load the queried data into a  [Pydantic](https://docs.pydantic.dev/latest/) model. It requires manual parsing and loading into a dictionary similar to the model before loading or manually assigning data to the model. Yes, it is doable but inconvenient.

## Desired Behavior
The preferred outcome is a way to get a dictionary object that can be loaded into a user-defined model. This will add convenience to the end user and not require that Pydantic be added to `filequery`, removing the need to add another dependency.

### Example
```python
from pydantic import BaseModel

class UserModel(BaseModel):
    name: str
    age: int


def load_query_to_model():
    # Query files loaded into db
    user_query = filequery.run(query)

    user_model = UserModel(**user_query.dict())
    return user_model
```

## Additional Context
My team and I use SQLModel for our ORM based on Pydantic for data validation. Our company uses CSVs as the backbone of data transfer between departments, and `filequery` has provided a simple way to get the needed data from the files, but it is a little unwieldy to parse the data into our models.

I would be happy to dig in and find a way to implement this if needed.

**Comment by MarkyMan4** [comment]
Hello! Thanks for opening the issue. I'm glad filequery has been helpful for you and your team.

I'm not very familiar with Pydantic, so just trying to understand how you would need the dictionary formatted. For example, if the query result was this:

```
col1,col2,col3
1,test 1,0.1
2,test 2,0.2
```

Would you be looking for a way to get a list of dictionaries where each element is a row from the query result like this?

```
[
    {
        "col1": 1,
        "col2": "test 1",
        "col3": 0.1
    },
    {
        "col1": 2,
        "col2": "test 2",
        "col3": 0.3
    },
]
```

Let me know if I'm understanding this correctly. I think we could do something in the `QueryResult` class to format the data how you need.

**Comment by derpy4me** [comment]
Looking back I was stuck in the notion of a way to tightly integrate Pydantic with Filequery.
Instead a simpler and ubiquitous method would be to retrieve/export the queried data in JSON format. Should be simpler and allow the data results to be used across any application or library that supports JSON (which Pydantic does).

**Comment by MarkyMan4** [comment]
I was just trying out a few things, and one thing that would be very easy to add is the ability to export a query to a JSON file. Right now it only supports CSV and Parquet exports (I guess I forgot to add the JSON option!). So with this change, you would be able to do something like this:

```python
from filequery.filedb import FileDb, FileType

fdb = FileDb('example/test.csv')
fdb.export_query('select * from test', 'result.json', FileType.JSON)
```

Which would create a file `result.json` that looks like this:

```json
[
	{"col1":1,"col2":"test 1","col3":0.1},
	{"col1":2,"col2":"test 2","col3":0.2}
]
```

I'll include this for sure in the next release since it's such a small change, but do you think it would still be helpful to be able to get a list of dictionaries from a `QueryResult` object?

**Comment by derpy4me** [comment]
Yes, I think that would be awesome.

**Comment by MarkyMan4** [comment]
I just released v0.2.4 which should address what we talked about here. #45 has the changes I included for this. You have a couple options now. You can export the a query to a JSON file like I showed above, or you can get a list of dictionaries using the `dict_records` property I added to the `QueryResult` class:

```python
from filequery.filedb import FileDb

fdb = FileDb("test.csv")
res = fdb.exec_query("select * from test")

print(res.dict_records)

# output: [{'col1': 1, 'col2': 'test 1', 'col3': 0.1}, {'col1': 2, 'col2': 'test 2', 'col3': 0.2}]
```

Let me know if you run into any issues. Appreciate your help on this!
